api/views.py

```diff
@@ -0,0 +1,176 @@
+    serializer_class = IncomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_incomes(self, request, *args, **kwargs):
+        number_of_incomes_to_show = request.GET.get('number_of_incomes', None)
+        if number_of_incomes_to_show is not None:
+            incomes = Income.objects.order_by('date')[:int(number_of_incomes_to_show)]
+        else:
+            incomes = Income.objects.order_by('date')
+        serializer = EventSerializer(incomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum_of_cost(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        current_date = datetime.today().date()
+        date_for_filter = current_date - timedelta(days=int(days_back))
+        incomes = Income.objects.filter(date__gte=date_for_filter)
+        sum = 0
+        for income in incomes:
+            sum += int(income.sum)
+        return Response("Total income from the last " + str(days_back) + " days is: " + str(sum))
+
+
+class OutcomeViewSet(viewsets.ModelViewSet):
+    queryset = Outcome.objects.all()
+    serializer_class = OutcomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_outcomes(self, request, *args, **kwargs):
+        number_of_outcomes_to_show = request.GET.get('number_of_outcomes', None)
+        if number_of_outcomes_to_show is not None:
+            outcomes = Outcome.objects.order_by('date')[:int(number_of_outcomes_to_show)]
+        else:
+            outcomes = Outcome.objects.order_by('date')
+        serializer = EventSerializer(outcomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum_of_cost(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        current_date = datetime.today().date()
+        date_for_filter = current_date - timedelta(days=int(days_back))
+        outcomes = Outcome.objects.filter(date__gte=date_for_filter)
+        sum = 0
+        for outcome in outcomes:
+            sum += int(outcome.sum)
+        return Response("Total outcome from the last " + str(days_back) + " days is: " + str(sum))
+
+
+class EventViewSet(viewsets.ModelViewSet):
+    queryset = Event.objects.all()
+    serializer_class = EventMiniSerializer
+
+    def get_future_events(self):
+        events = Event.objects.filter(date__gte=datetime.today().date())
+        current_date = datetime.today().date()
+        current_time = datetime.today().time()
+        for event in events:
+            if event.date == current_date:
+                if event.time < current_time:
+                    events = events.exclude(id=event.id)
+        return events
+
+    @action(detail=False, methods=['get'])
+    def future_events(self, request, *args, **kwargs):
+        events = self.get_future_events().order_by('date', 'time')
```

Comment:
As a side note, I would implement this method on the QuerySet level, using custom managers and custom querysets: https://docs.djangoproject.com/en/2.1/topics/db/managers/

Then you would be able to do it simply like this: `Events.objects.future_events()` or `Events.future_events()`. You could also implement `past_events()` and `today_events()` very simply.

Reply:
done